source/Plugins/Language/Swift/SwiftDictionary.cpp

```diff
@@ -26,95 +26,48 @@ using namespace lldb_private;
-public:
-  SwiftHashedContainerBufferHandler::Kind GetKind() {
-    return Kind::eDictionary;
-  }
-
-  static ConstString GetMangledStorageTypeName();
-
-  static ConstString GetDemangledStorageTypeName();
-
-  SwiftDictionaryNativeBufferHandler(ValueObjectSP nativeStorage_sp,
-                                     CompilerType key_type,
-                                     CompilerType value_type)
-      : SwiftHashedContainerNativeBufferHandler(nativeStorage_sp, key_type,
-                                                value_type) {}
-  friend class SwiftHashedContainerBufferHandler;
-
-private:
-};
-
-class SwiftDictionarySyntheticFrontEndBufferHandler
-    : public SwiftHashedContainerSyntheticFrontEndBufferHandler {
-public:
-  SwiftHashedContainerBufferHandler::Kind GetKind() {
-    return Kind::eDictionary;
-  }
-
-  virtual ~SwiftDictionarySyntheticFrontEndBufferHandler() {}
-
-  SwiftDictionarySyntheticFrontEndBufferHandler(lldb::ValueObjectSP valobj_sp)
-      : SwiftHashedContainerSyntheticFrontEndBufferHandler(valobj_sp) {}
-  friend class SwiftHashedContainerBufferHandler;
-
-private:
-};
-
-class DictionarySyntheticFrontEnd : public HashedContainerSyntheticFrontEnd {
-public:
-  DictionarySyntheticFrontEnd(lldb::ValueObjectSP valobj_sp)
-      : HashedContainerSyntheticFrontEnd(valobj_sp) {}
-
-  virtual bool Update();
-
-  virtual ~DictionarySyntheticFrontEnd() = default;
-};
-}
-}
-}
-
-SyntheticChildrenFrontEnd *
-lldb_private::formatters::swift::DictionarySyntheticFrontEndCreator(
-    CXXSyntheticChildren *, lldb::ValueObjectSP valobj_sp) {
-  if (!valobj_sp)
-    return NULL;
-  return (new DictionarySyntheticFrontEnd(valobj_sp));
+const DictionaryConfig &
+DictionaryConfig::Get() {
+  static DictionaryConfig g_config{};
+  return g_config;
 }
 
-bool lldb_private::formatters::swift::DictionarySyntheticFrontEnd::Update() {
-  m_buffer = SwiftHashedContainerBufferHandler::CreateBufferHandler(
-      m_backend,
-      [](ValueObjectSP a, CompilerType b,
-         CompilerType c) -> SwiftHashedContainerBufferHandler * {
-        return new SwiftDictionaryNativeBufferHandler(a, b, c);
-      },
-      [](ValueObjectSP a) -> SwiftHashedContainerBufferHandler * {
-        return new SwiftDictionarySyntheticFrontEndBufferHandler(a);
-      },
-      SwiftDictionaryNativeBufferHandler::GetMangledStorageTypeName(),
-      SwiftDictionaryNativeBufferHandler::GetDemangledStorageTypeName());
-  return false;
+DictionaryConfig::DictionaryConfig()
+  : HashedCollectionConfig() {
+  m_summaryProviderName =
+    ConstString("Swift.Dictionary summary provider");
+  m_syntheticChildrenName =
+    ConstString("Swift.Dictionary synthetic children");
+  m_collection_demangledRegex =
+    ConstString("^Swift\\.Dictionary<.+,.+>$");
+
+  // Native storage class
+  m_nativeStorage_mangledPrefix =
+    ConstString(SwiftLanguageRuntime::GetCurrentMangledName("_TtGCs37_HashableTypedNativeDictionaryStorage"));
```

Comment:
You can probably use the new mangled name here.

Reply:
We apparently still use this mangling for the Objective-C runtime names of NSObject subclasses defined in Swift.

The name is incomplete, so while I cargo-culted GetCurrentMangledName over from the old code, it actually has no effect.